src/bp/ui-studio/src/web/components/Layout/StatusBar/NluPerformanceStatus.tsx

```diff
@@ -0,0 +1,119 @@
+import { Colors } from '@blueprintjs/core'
+import axios from 'axios'
+import React from 'react'
+import { FaThLarge } from 'react-icons/fa'
+
+import ActionItem from './ActionItem'
+import style from './StatusBar.styl'
+
+const GREEM_LOWER_BOUND = 0.85
+const RED_UPPER_BOUND = 0.5
+
+interface MatrixInfo {
+  matrix: any
+  confusionComputing: boolean
+}
+
+interface Props {
+  synced: boolean
+  updateSyncStatus: (synced: boolean) => void
+}
+
+interface State {
+  f1: number
+  synced: boolean
+  computing: boolean
+}
+
+export default class NluPerformanceStatus extends React.Component<Props, State> {
+  state: State = {
+    f1: undefined,
+    synced: true,
+    computing: false
+  }
+
+  componentDidMount() {
+    // tslint:disable-next-line: no-floating-promises
+    this.fetchConfusion()
+  }
+
+  componentDidUpdate() {
+    if (!this.props.synced && this.state.synced) {
+      this.setState({ f1: undefined, synced: false })
+    }
+  }
+
+  fetchConfusion = async (modelHash?: string) => {
+    const { matrix, confusionComputing } = await this.getConfusionMatrix(modelHash)
+    const f1 = confusionComputing ? undefined : this.extractF1FromMatrix(matrix)
+    const synced = !!matrix
+    this.props.updateSyncStatus(synced)
+    this.setState({ f1, synced: !!matrix, computing: confusionComputing })
+  }
+
+  getConfusionMatrix: (modelHash?: string) => Promise<MatrixInfo> = async (modelHash?) => {
+    if (!modelHash) {
+      const modelHashResponse = await axios.get(`${window.BOT_API_PATH}/mod/nlu/currentModelHash`)
+      modelHash = modelHashResponse.data
+    }
+
+    try {
+      const confusionMatrixResponse = await axios.get(`${window.BOT_API_PATH}/mod/nlu/confusion/${modelHash}`)
+      return confusionMatrixResponse.data as MatrixInfo
+    } catch {
+      return { matrix: undefined, confusionComputing: false }
+    }
+  }
+
+  extractF1FromMatrix(matrix): number {
+    if (!matrix || !matrix.intents || !matrix.intents.all) {
+      return undefined
+    }
+    return matrix.intents.all.f1
+  }
+
+  computationMutex = false // not in the state because it needs to be updated in sync
+  calculateConfusion = async () => {
+    const { confusionComputing } = await this.getConfusionMatrix()
+    this.computationMutex = this.computationMutex || confusionComputing
+
+    if (!this.computationMutex) {
+      this.computationMutex = true
+      this.setState({ f1: undefined, computing: true })
+
+      const response = await axios.post(`${window.BOT_API_PATH}/mod/nlu/confusion`)
+
+      this.computationMutex = false
+      this.setState({ computing: false })
+      await this.fetchConfusion(response.data.modelHash)
+    }
+  }
+
+  mapF1ToColor(f1: number) {
```

Comment:
Typings are a bit off, it should be more like mapF1ToColor(f1: number | undefined): string